src/weapon.py

```diff
@@ -39,22 +40,20 @@ def __init__(self, input_button: VirtualButton, car: arcade.Sprite):
         self.beam_projection = arcade.SpriteSolidColor(self.beam_range, 5, arcade.color.RED)
         #this is currently used to store half the lenght so the update method can make the beam not centered on the car
         self.beam_projection.velocity = self.beam_range/2
-        self.button_held = False
 
     def update(self, delta_time, projectile_list: arcade.SpriteList, beam_list: arcade.SpriteList):
         if self.input_button.pressed:
             self.shoot(beam_list)
-        if self.button_held:
-            if not self.input_button.value:
-                self.remove_bullets(beam_list)
+        if self.input_button.released and self.beam_projection in beam_list:
+            beam_list.remove(self.beam_projection)
 
     def shoot(self, beam_list: arcade.SpriteList):
-        self.button_held = True
         beam_list.append(self.beam_projection)
 
-    def remove_bullets(self, beam_list: arcade.SpriteList):
-        beam_list.remove(self.beam_projection)
-        self.button_held = False
+    def swap_out(self, beam_list: arcade.SpriteList):
+        if self.beam_projection in beam_list:
+            beam_list.remove(self.beam_projection)
```

Comment:
This is duplicated in `def update`; it's only 2 lines, but could be extracted to its own method to DRY things up.